contracts/SimpleGAMultiSig.aes

```diff
@@ -78,38 +89,57 @@ contract SimpleGAMultiSig =
       Some(tx) =>
         require(is_expired(tx), "ERROR_A_TX_IS_ALREADY_PROPOSED")
         set_current_tx(ga_tx_hash, ttl)
-  
+
   // signer confirms tx
   stateful entrypoint confirm(sig : signature, n : int) =
     check_and_abort_or_update_nonce(n)
     let tx = Option.force_msg(state.current_tx, "ERROR_NOTHING_TO_CONFIRM")
     require_valid_signer_and_sig(Call.caller, tx.ga_tx_hash, sig)
-    require(!List.contains(Call.caller, tx.confirmed_by),"ERROR_ALREADY_CONFIRMED")
+    require(!Set.member(Call.caller, tx.confirmed_by),"ERROR_ALREADY_CONFIRMED")
     require(!is_expired(tx), "ERROR_TX_ALREADY_EXPIRED")
-    let updated_tx = tx{confirmed_by=Call.caller :: tx.confirmed_by}
+    let updated_tx = tx{confirmed_by=Set.insert(Call.caller, tx.confirmed_by)}
```

Comment:
if its map instead of set the syntax would be nicer here probably

Reply:
I still think set makes more sense than map